frontend/apps/crates/entry/module/memory/play/src/base/state.rs

```diff
@@ -53,18 +53,36 @@ impl Base {
             ..
         } = init_args;
 
-        let content = raw.content.unwrap_ji();
+        let mut rng = thread_rng();
 
-        let n_cards = content.base.pairs.len() * 2;
+        let mut content = raw.content.unwrap_ji();
+
+        let max_cards = content.player_settings.pairs_to_display
+            .map_or_else(|| {
+                let card_count = content.base.pairs.len();
+                if card_count > MAX_LIST_WORDS {
+                    MAX_LIST_WORDS
+                } else {
+                    card_count
+                }
+            }, |pairs_to_display| {
+                if pairs_to_display > MAX_LIST_WORDS as u32 {
+                    MAX_LIST_WORDS
+                } else {
+                    pairs_to_display as usize
+                }
+            });
+
+        let n_cards = max_cards * 2;
         let mut pair_lookup: Vec<usize> = vec![0; n_cards];
         let mut cards = {
-            let pairs = &content.base.pairs;
+            let pairs = &mut content.base.pairs;
+            pairs.shuffle(&mut rng);
```

Comment:
Shuffle the pair so that if there are more pairs than the max or than specified in the setting, each time a student plays the game, a different set of pairs should be rendered